askama_derive/src/generator.rs

```diff
@@ -309,7 +383,14 @@ impl<'a> Generator<'a> {
         heritage: Option<&'n Heritage<'_>>,
         locals: MapChain<'n, &'n str, LocalMeta>,
         whitespace: WhitespaceHandling,
+        discard_output: bool,
     ) -> Generator<'n> {
+        let buf_writable = if discard_output {
+            WritableBuffer::new_discarding()
+        } else {
+            WritableBuffer::new()
+        };
```

Comment:
Please use a `match` expression for this, and inline it into the initializer expression directly instead of binding here. Let's ditch the constructors for `WritableBuffer` and initialize its fields directly here (and rename the argument to the `Generator::new()` constructor as just `discard` to match).